### PR TITLE
Vary the size of the inputs for the benchmarks

### PR DESCRIPTION
**Pull request by bowlofeggs**
Signed-off-by: Randy Barlow <randy@electronsweatshop.com>

### CODE REVIEW

File: benches/all.rs [hunk 1]
```diff
@@ -7,58 +9,78 @@ use primes::{
...
-        b.iter(|| {
-            let _: Vec<u32> = Primes::new().take(LARGEST_PRIME_INDEX).collect();
-        })
-    });
+fn bench_em(c: &mut Criterion) {
+    let mut group = c.benchmark_group("Primes");
+    // For each tuple, the first number is the largest prime number we will search for for
+    // eratosthenes-based algorithms and the second number is the index of that same prime
+    // in an array of prime numbers.
+    for i in [10_000, 100_000, 1_000_000, 10_000_000, 100_000_000, 500_000_000].iter() {
```
Comment (bowlofeggs) [inline benches/all.rs]: Running all of these does take hours, but I've confirmed that you can ask at the CLI to run a subset of them pretty easily.

e.g., you can ask for a specific test like this:

```
$ cargo bench -- live_first_prime/1000000\$
```

Or, you can run all the 10k ones like this:

```
$ cargo bench -- 10000\$
```

File: benches/all.rs [hunk 2]
```diff
@@ -7,58 +10,116 @@ use primes::{
...
 
-fn test_division(c: &mut Criterion) {
-    c.bench_function("test_division", |b| {
-        b.iter(|| {
-            let _: Vec<u32> = Primes::new().take(LARGEST_PRIME_INDEX).collect();
-        })
-    });
+fn bench_em(c: &mut Criterion) {
+    let mut group = c.benchmark_group("Primes");
+    // For each tuple, the first number is the largest prime number we will search for for
```
Comment (andrewhalle) [inline benches/all.rs]: Is this comment still applicable? It mentions tuples, which I don't see.

**Review by andrewhalle — CHANGES_REQUESTED** [review]
One minor thing

File: benches/all.rs (hunk 2 above)
Comment (andrewhalle) [reply benches/all.rs]: I think this comment is pre-`Input` so might be better phrased as "Each number in this list defines a search space in which we will record the largest prime and the index of that prime" or similar

File: benches/all.rs [hunk 3]
```diff
@@ -7,58 +10,116 @@ use primes::{
...
 
-criterion_group!(
-    all,
-    live_first_primes,
-    live_second_primes,
-    test_division,
-    sieve_og,
-    sieve_skip_2
-);
+criterion_group!(all, bench_em,);
```
Comment (andrewhalle) [inline benches/all.rs]: Doesn't hurt anything, but I think the trailing comma can be removed as well, that's how this macro appears in the criterion docs.

**Review by andrewhalle — CHANGES_REQUESTED** [review]

**Review by andrewhalle — APPROVED** [review]

**Pull request merged**